cli/command/context/list.go

```diff
@@ -76,7 +76,14 @@ func runList(dockerCli command.Cli, opts *listOptions) error {
 	sort.Slice(contexts, func(i, j int) bool {
 		return sortorder.NaturalLess(contexts[i].Name, contexts[j].Name)
 	})
-	return format(dockerCli, opts, contexts)
+	if err := format(dockerCli, opts, contexts); err != nil {
+		return err
+	}
+	if isDockerHostSet() {
+		fmt.Fprint(dockerCli.Err(), "Warning: DOCKER_HOST environment variable overrides the active context. "+
+			"To use a context, either set the global --context flag, or unset DOCKER_HOST environment variable.\n")
```

Comment:
Wondering (orthogonal); could we have a `--force` flag to unset `DOCKER_HOST` ? i.e.;

```bash
export DOCKER_HOST=foo
docker context use --force my-context

echo $DOCKER_HOST
# not set
```

Reply:
How would you unset an env variable in the parent process?

Reply:
LOL, good point; magic?

![magic](https://media.giphy.com/media/12NUbkX6p4xOO4/giphy.gif)

Reply:
cc @chris-crone !